### ISSUE TITLE
Ghost version using canary release

### ISSUE THREAD

**Issue by Carduelis** [lerna/lerna#2778]
Hello!

We're trying to use `--canary` release, to publish new package version with hash of commit in the end.
Unfortunately, we get some old version which lerna produces.
```
lerna publish --canary --preid "${CI_BUILD_REF_NAME}-$(echo ${CI_BUILD_REF} | cut -c-8)" --registry ${NPM_REGISTRY} --yes
````
This command produces the following output:
```
Found 3 packages to publish:
  - @myscope/uikit => 0.27.3-branch-name-e694052b.120+e694052b
  - @myscope/storybook => 0.27.3-branch-name-e694052b.120+e694052b
  - @myscope/theme => 0.27.3-branch-name-e694052b.120+e694052b
```
_(we added commit hash, cause the hash after `+` sign is trimmed when published to a private registry. There is an issue for that already.)_

```
// lerna.json
{
  "command": {
    "version": {
      "message": "[skip ci] chore(release): release %s",
      "allowBranch": ["master"]
    }
  },
  "packages": ["packages/**"],
  "version": "0.36.5"
}
```

The whole project doesn't have anything related to `0.27.2` or `0.27.3`. All `package.json` files have `0.36.5` version in it.

I tried to check the source code of `publish --canary`, found this:
```
const makeVersion = fallback => ({ lastVersion = fallback, refCount, sha }) => {
      // the next version is bumped without concern for preid or current index
      const nextVersion = semver.inc(lastVersion.replace(this.tagPrefix, ""), release.replace("pre", ""));

      // semver.inc() starts a new prerelease at .0, git describe starts at .1
      // and build metadata is always ignored when comparing dependency ranges
      return `${nextVersion}-${preid}.${Math.max(0, refCount - 1)}+${sha}`;
    };
```
https://github.com/lerna/lerna/blob/f6e7a13e60fefc523d701efddfcf0ed41a77749b/commands/publish/index.js#L359

But still have no idea where `0.27.3` comes from.

**Expected behavior** is to have:

```
Found 3 packages to publish:
  - @myscope/uikit => 0.36.6-branch-name-e694052b.120+e694052b
  - @myscope/storybook => 0.36.6-branch-name-e694052b.120+e694052b
  - @myscope/theme => 0.36.6-branch-name-e694052b.120+e694052b
```

Cause we have 
```
{
   "version": "0.36.5"
}
```

**Comment by reallifeliver** [comment]
I have simmiar issue. my package's version is 0.5.0.
but the pre-released version is 0.0.39-next.1

I don't know why this happen

**Comment by ChiefORZ** [comment]
duplicate of https://github.com/lerna/lerna/issues/2622
and we fixed is with following changes in our forked lerna: https://github.com/lerna/lerna/pull/2572
